### PR TITLE
SearchV2: move search parameters to a nested object

### PR DESCRIPTION
**Pull request by ryantxu**
This cleans up the search interface so that query parameters exist as a nested object in the `DataQuery` rather than as values on the top level.

To test:
* run grafana in dev mode w/ alpha panels enabled
* add a panel to a dashboard using the "-- Grafana --" datasource
* select "search" type

### CODE REVIEW

File: public/app/plugins/datasource/grafana/datasource.ts [hunk 1]
```diff
@@ -85,13 +85,6 @@ export class GrafanaDatasource extends DataSourceWithBackend<GrafanaQuery> {
         let channel = templateSrv.replace(target.channel, request.scopedVars);
         const { filter } = target;
 
-        // Help migrate pre-release channel paths saved in dashboards
-        // NOTE: this should be removed before V8 is released
-        if (channel && channel.startsWith('telegraf/')) {
```
Comment (ryantxu) [inline public/app/plugins/datasource/grafana/datasource.ts:90]: should likely be in a different PR.. but I stumbled into it here, and did not want to lose it!

File: public/app/plugins/datasource/grafana/components/SearchEditor.tsx [hunk 2]
```diff
@@ -0,0 +1,78 @@
...
+  const handleSearchEnterKey = (e: React.KeyboardEvent<HTMLInputElement>) => {
+    if (e.key !== 'Enter') {
+      return;
+    }
+    handleSearchBlur(e as any);
+  };
+
+  const onSaveSearchJSON = (rawSearchJSON: string) => {
+    try {
+      onChange(JSON.parse(rawSearchJSON) as SearchQuery);
```
Comment (ArturWierzbicki) [inline public/app/plugins/datasource/grafana/components/SearchEditor.tsx:47]:
```suggestion
      const searchQuery = JSON.parse(rawSearchJSON) as SearchQuery;
      onChange(searchQuery);
      setQuery(searchQuery.query ?? '');
```

File: public/app/plugins/datasource/grafana/components/SearchEditor.tsx [hunk 3]
```diff
@@ -0,0 +1,78 @@
...
+  onChange: (value: SearchQuery) => void;
+}
+
+export default function SearchEditor({ value, onChange }: Props) {
+  const [json, setJSON] = useState('');
+  const [query, setQuery] = useState(value.query ?? '');
+
+  useEffect(() => {
+    const emptySearchQuery: SearchQuery = {
+      query: '*',
```
[the remark this answers is not in the document]
Comment (ryantxu) [reply public/app/plugins/datasource/grafana/components/SearchEditor.tsx:17]: yes, but lets do that in a different PR.  This one just moves the object position, and does not change any behavior

**Review by ArturWierzbicki — APPROVED** [review]
🚀
